apps/api/src/billing/services/chain-error/chain-error.service.ts

```diff
@@ -54,12 +65,52 @@ export class ChainErrorService {
     }
 
     const messagePrefix = this.getMessagePrefix(error, messages);
-    const { message, code } = this.ERRORS[clue];
+
+    const { message, code } = (await this.getBalanceError(clue, error.message)) || this.ERRORS[clue];
     const prefixedMessage = messagePrefix ? `${messagePrefix}: ${message}` : message;
 
     return createError(code, prefixedMessage, { originalError: error });
   }
 
+  private async getBalanceError(clue: string, message: string) {
+    if (clue !== "insufficient funds") return;
+
+    const masterWalletAddress = await this.masterWallet.getFirstAddress();
+    const { amount } = await this.balanceHttpService.getBalance(masterWalletAddress, this.billingConfigService.get("DEPLOYMENT_GRANT_DENOM"));
+    const parsedMessage = this.parseInsufficientFundsErrorMessage(message);
+
+    if (parsedMessage && amount < parsedMessage.requiredAmount) {
+      return {
+        code: 503,
+        message: "Service temporarily unavailable"
+      };
+    }
+  }
+
+  private parseInsufficientFundsErrorMessage(message: string): {
+    availableAmount: number;
+    requiredAmount: number;
+    denomination: string;
+  } | null {
+    const usdcDenoms = Object.values(this.billingConfigService.get("USDC_IBC_DENOMS"))
+      .map(denom => denom.replace(/\//g, "\\/"))
+      .join("|");
+
+    const pattern = new RegExp(`(\\d+)(uakt|${usdcDenoms}) is smaller than (\\d+)\\2`);
+
+    const match = message.match(pattern);
+
+    if (!match) {
+      return null;
+    }
+
+    return {
+      availableAmount: parseInt(match[1], 10),
+      requiredAmount: parseInt(match[3], 10),
+      denomination: match[2]
```

Comment:
```suggestion
      denom: match[2]
```
just coz we use this term across the app

Reply:
SGTM, done